handshake/stk_generator.go

```diff
@@ -0,0 +1,72 @@
+package handshake
+
+import (
+	"bytes"
+	"net"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/crypto"
+)
+
+// An STK is a source address token
+type STK struct {
+	RemoteAddr string
+	SentTime   time.Time
+}
+
+// An STKGenerator generates STKs
+type STKGenerator struct {
```

Comment:
The one in the handshake package deals with the net.Addr of the client and decides how to encode this. The one in the crypto package encodes arbitrary byte slices into an STK.